packages/grid_client/scripts/casperlabs.ts

```diff
@@ -0,0 +1,97 @@
+import { FilterOptions, MachinesModel } from "../src";
+import { config, getClient } from "./client_loader";
+import { log } from "./utils";
+
+async function deploy(client, vms) {
+  try {
+    const res = await client.machines.deploy(vms);
+    log("================= Deploying casperlabs =================");
+    log(res);
+    log("================= Deploying casperlabs =================");
+  } catch (error) {
+    log("Error while Deploying the VM " + error);
+  }
+}
+
+async function getDeployment(client, vms) {
+  try {
+    const res = await client.machines.getObj(vms);
+    log("================= Getting deployment information =================");
+    log(res);
+    log(`You can access casperlabs via the browser using: http://newVMS5.${res[0].env.CASPERLABS_WEBSERVER_HOSTNAME}`);
+    log("================= Getting deployment information =================");
+  } catch (error) {
+    log("Error while getting the deployment " + error);
+  }
+}
+
+async function cancel(client, vms) {
+  try {
+    const res = await client.machines.delete(vms);
+    log("================= Canceling the deployment =================");
+    log(res);
+    log("================= Canceling the deployment =================");
+  } catch (error) {
+    log("Error while canceling the deployment " + error);
+  }
+}
+
+async function main() {
+  const grid3 = await getClient();
+
+  const vmQueryOptions: FilterOptions = {
+    cru: 4,
+    mru: 4, // GB
+    sru: 10,
+    farmId: 1,
+  };
+
+  const vms: MachinesModel = {
+    name: "newVMS5",
+    network: {
+      name: "wedtest",
+      ip_range: "10.249.0.0/16",
+    },
+    machines: [
+      {
+        name: "testvm",
+        node_id: +(await grid3.capacity.filterNodes(vmQueryOptions))[0].nodeId,
+        disks: [
+          {
+            name: "wedDisk",
+            size: 8,
+            mountpoint: "/var/lib/docker",
+          },
+        ],
+        public_ip: true,
+        public_ip6: false,
+        planetary: false,
+        mycelium: true,
+        cpu: 4,
+        memory: 1024 * 4,
+        rootfs_size: 0,
+        flist: "https://hub.grid.tf/tf-official-apps/casperlabs-latest.flist",
+        entrypoint: "/sbin/zinit init",
+        env: {
+          SSH_KEY: config.ssh_key,
+          CASPERLABS_HOSTNAME: "gent01.dev.grid.tf",
+        },
+      },
+    ],
+    metadata: "",
+    description: "casperlabs machine",
+  };
+
+  //Deploy casperlabs
+  await deploy(grid3, vms);
+
+  //Get the deployment
+  await getDeployment(grid3, vms.name);
+
+  //Uncomment the line below to cancel the deployment
+  await cancel(grid3, { name: vms.name });
+
+  // await grid3.disconnect();
```

Comment:
The connection hangs as this line is commented.

```suggestion
  await grid3.disconnect();
```